### ISSUE TITLE
Unhelpful hint in new javascript course, platformer game, step 46

### ISSUE THREAD

**Issue by DBailey0629** [freeCodeCamp/freeCodeCamp#53962]
### Describe the Issue

The hint just repeats the instruction. An appropriate hint would be to remind the camper that the "this" keyword is used in the construction of the Player class and now we have a player object.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-intermediate-oop-by-building-a-platformer-game/step-46

### Your code

```

  if(keys.rightKey.pressed && player.position.x < proportionalSize(400)){}


```

### Expected behavior

When camper submits an incorrect response using the "this" keyword instead of "player", the hint should update to remind them that "this" is used to construct the class called by the object "player".

### System

- Device: Laptop
- OS: macOS
- Browser: Chrome
- Version: current

**Comment by PrajvalaRC** [comment]
@DBailey0629, 

So you are expecting mentioned hint when user codes as below,
if(keys.rightKey.pressed && this.position.x < proportionalSize(400)){}

**Comment by DBailey0629** [comment]
@PrajvalaRC ,
Yes, that is probably the best. Beyond this one step, I also encountered a similar difficulty on multiple steps in this lesson where the hint was just an almost exact restatement of the instruction. Some of them seemed to be a too strict check on requiring parentheses around a single parameter for a callback function.

**Comment by Atyagi1213** [comment]
as far as your concern on the matter of strict checking on requiring a parenthesis around a single parameter for a callback function, it is because it is a good practice in general when using JavaScript and also adopted by Freecodecamp . you can read them here https://forum.freecodecamp.org/t/free-code-camp-javascript-style-guide/19121 . If you don't want to put curly braces in single line functions, you can opt for arrow functions.

**Comment by Atyagi1213** [comment]
> @PrajvalaRC , Yes, that is probably the best. Beyond this one step, I also encountered a similar difficulty on multiple steps in this lesson where the hint was just an almost exact restatement of the instruction. Some of them seemed to be a too strict check on requiring parentheses around a single parameter for a callback function.

On the matter of providing more helpful hints I understand your concern. So basically, you want to know what is wrong with the code you wrote and why it doesn't work and simultaneously providing the correct solution and explanation. Am I understanding it clearly?

**Comment by jdwilkin4** [comment]
As for the hint, we could go something with this:

```md
You should have an `if` statement that checks if the right key was pressed and the player's `x` position is less than `proportionalSize(400)`. The `this` keyword should not be used here because that is only for the `Player` class and not the use of the `player` object.
```

**Comment by jdwilkin4** [comment]
@Atyagi1213 

> as far as your concern on the matter of strict checking on requiring a parenthesis around a single parameter for a callback function, it is because it is a good practice in general when using JavaScript and also adopted by Freecodecamp 

Beginners shouldn't be penalized for incorrect answers because of differences in parenthesis though.
Unless it is a syntax issue, if the beginner is writing technically correct code, then the test should pass.

Those changes should be handled outside of the scope of this issue though

**Comment by Nitz2611** [comment]
@jdwilkin4 

> As for the hint, we could go something with this:
> 
> ```
> You should have an `if` statement that checks if the right key was pressed and the player's `x` position is less than `proportionalSize(400)`. The `this` keyword should not be used here because that is only for the `Player` class and not the use of the `player` object.
> ```

what if a user didn't use the 'this' keyword but made a different mistake, should the hint related to the 'this' keyword still be included?

**Comment by PrajvalaRC** [comment]
> @PrajvalaRC , Yes, that is probably the best. Beyond this one step, I also encountered a similar difficulty on multiple steps in this lesson where the hint was just an almost exact restatement of the instruction. Some of them seemed to be a too strict check on requiring parentheses around a single parameter for a callback function.

@DBailey0629 
If we have to be that specific, won't we have to add many cases to handle it?

**Comment by PrajvalaRC** [comment]
> > As for the hint, we could go something with this:
> > ```
> > You should have an `if` statement that checks if the right key was pressed and the player's `x` position is less than `proportionalSize(400)`. The `this` keyword should not be used here because that is only for the `Player` class and not the use of the `player` object.
> > ```
> 
> what if a camper didn't use the 'this' keyword but made a different mistake, should the hint related to the 'this' keyword still be included?

Instead of adding above statement in hints, it might be better option to add in description as general 'Note'. Else it might confuse user.

**Comment by jdwilkin4** [comment]
@PrajvalaRC 

A note in addition to the updates to the hint would work.
Remember that unfortunately, most people don't read everything.
So if the camper skips the note, then it is reiterated in the hint

**Comment by Atyagi1213** [comment]
> @Atyagi1213
> 
> > as far as your concern on the matter of strict checking on requiring a parenthesis around a single parameter for a callback function, it is because it is a good practice in general when using JavaScript and also adopted by Freecodecamp
> 
> Beginners shouldn't be penalized for answers because of differences in parenthesis though. Unless it is a syntax issue, if the beginner is writing technically correct code, then the test should pass.
> 
> Those changes should be handled outside of the scope of this issue though

The problem stated by @DBailey0629 regarding the parenthesis is a syntax error because only arrow functions allow parameters without brackets . So technically her code will become incorrect .
